src/Module/SessionRuleFactory.php

```diff
@@ -109,10 +109,11 @@ public function make($config = null)
         $repeatUntilPeriod    = ($repeatUntilPeriod < 1) ? 1 : $repeatUntilPeriod;
         $repeatUntilPeriodStr = sprintf('+%1$d %2$s', $repeatUntilPeriod - 1, $repeatUnit);
         $repeatUntilDate      = $this->_containerGet($config, 'repeat_until_date');
-        // Calculate the end of repetition
-        $repeatEnd = ($repeatUntil === 'period')
-            ? Carbon::createFromTimestampUTC($start)->modify($repeatUntilPeriodStr)->getTimestamp()
-            : $repeatUntilDate;
+        // Calculate the end of repetition datetime
+        $repeatEndDt = ($repeatUntil === 'period')
+            ? Carbon::createFromTimestampUTC($start)->modify($repeatUntilPeriodStr)
+            : Carbon::createFromTimestampUTC($repeatUntilDate)->addDay(1);
```

Comment:
Shouldn't you subtract 1 second from there?

Reply:
End time is exclusive. Otherwise I would have added 23 hours, 59 minutes and 59 seconds.

Reply:
For reference, [this is why](https://github.com/RebelCode/rcmod-eddbk-session-generator/blob/develop/src/Generator/AbstractIteratorRule.php#L274) it's exclusive.